### PR TITLE
Fix price display when quantity discount on combination

### PR DESCRIPTION
**Pull request by erouvier29**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | Fix price display when quantity discount on combination
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/PSCSX-7797
| How to test?  | Cf ticket

### CODE REVIEW

**Comment by vincentbz** [comment]
Hi Eric !

Thanks for that PR, it works perfectly :)

**Comment by vincentbz** [comment]
We have to see if this PR is a duplicate of this one : https://github.com/PrestaShop/PrestaShop/pull/7249 and which one is the most relevant. Both seem to be working

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -710,7 +711,7 @@ protected function formTargetFormat()
         $this->context->smarty->assign('customizationFormTarget', $customization_form_target);
     }
 
-    protected function formatQuantityDiscounts($specific_prices, $price, $tax_rate, $ecotax_amount)
```
Comment (aleeks) [inline controllers/front/ProductController.php]: Maybe break some compatibility ? WDYT @maximebiloe

Comment (erouvier29) [inline controllers/front/ProductController.php]: You're right, there is some compatibility break. My rationale was: it is a protected function, then its signature can be changed. I just forgot inheritance, and especially overrides.
So, $price parameter could be kept as unused.
But anyway, first parameter $specific_prices should include additional base_price field in order the new formatting succeeds

Comment (maximebiloe) [inline controllers/front/ProductController.php]: Indeed, I agree with @aleeks, we can't change the function signature. Could you amend your commit to keep the parameter, please? Even if we won't use it anymore.

Regards

Comment (erouvier29) [inline controllers/front/ProductController.php]: Done

**Comment by maximebiloe** [comment]
Thank you @erouvier29

**Comment by kmorgen** [comment]
Unfortunately the Quantity Module is completely messed up in 1.6.1.13 now!
There is no relation between listed values on product page and what is actually calculated in the cart.
Changing the currency will make the values even worse!

Seems the listed values are calculated from default / retail price, but if quantity discount is made for a specific customer group (like resellers), the listings will be incorrect.

I have tried the following scenarios:
- created Fixed price for certain quantity and group = listing shows the discount value (difference in price), not the actual price anymore - but when added to cart calculation is correct.

- created a discount amount for a specific quantity and group = listing then shows the actual price (not the discount amount, so appears more correct), but wrong price is then calculated when added to cart.

If changing the currency in FO for both scenarios the listings will be completely messed up.

So basically, it does not work anymore...

**Comment by erouvier29** [comment]
Thanks @kmorgen  for your comment.
My fault, i didn't test wrt specific price for all combinations nor alternate currency.
You may have a look to the new PR #7869.
Regards

**Comment by kmorgen** [comment]
Hi Eric

 

Thanks for the response

 

I actually already created this Forge ticket: http://forge.prestashop.com/browse/PSCSX-9037

 

Your #7869 fix actually solves the product page listing (depending on currency). However please see the ticket for further description of the issue.

 

Med venlig hilsen

Kim Morgen  |   <http://www.morgen.dk/> www.morgen.dk  |  Tlf: 77 34 40 50

 

Fra: Eric Rouvier [mailto:notifications@github.com] 
Sendt: 7. maj 2017 07:58
Til: PrestaShop/PrestaShop <PrestaShop@noreply.github.com>
Cc: kmorgen <kim@morgen.dk>; Mention <mention@noreply.github.com>
Emne: Re: [PrestaShop/PrestaShop] Fix price display when quantity discount on combination (#7644)

 

Thanks @kmorgen <https://github.com/kmorgen>  for your comment.
My fault, i didn't test wrt specific price for all combinations nor alternate currency.
You may have a look to the new PR #7869 <https://github.com/PrestaShop/PrestaShop/pull/7869> .
Regards
